astropy/io/fits/convenience.py

```diff
@@ -512,6 +510,8 @@ def table_to_hdu(table, character_as_bytes=False):
 
         unit = table[col.name].unit
         if unit is not None:
+            from ...units import Unit
```

Comment:
Please add a comment as to why the imports are here - otherwise someone might move them back in future.

Reply:
@astrofrog - ok, done.